db-tests/src/test/java/gov/nasa/jpl/aerie/database/ActivityDirectiveChangelogTests.java

```diff
@@ -0,0 +1,310 @@
+                              );
+
+    res.next();
+    final var count = res.getInt(1);
+    res.close();
+
+    assertEquals(expectedCount, count);
+  }
+
+  @Test
+  void shouldCreateChangelogForInsertedActDir() throws SQLException {
+    final var expectedCount = 1;
+
+    final var planId = merlinHelper.insertPlan(
+        missionModelId,
+        merlinHelper.admin.name(),
+        "changelog plan" + UUID.randomUUID(),
+        "1970-1-1 00:00:00+00");
+    final var activityId = merlinHelper.insertActivity(planId);
+
+    final var res = connection.createStatement()
+                              .executeQuery(
+                                  // language=sql
+                                  """
+                                      SELECT count(revision)
+                                      FROM activity_directive_changelog
+                                      WHERE plan_id = %d and activity_directive_id = %d;"""
+                                      .formatted(planId, activityId)
+                              );
+
+    res.next();
+    final var revision = res.getInt(1);
+    res.close();
+
+    assertEquals(expectedCount, revision);
+  }
+
+  @Test
+  void shouldCreateChangelogForUpdatedActDir() throws SQLException {
+    final var expectedOffsetCurrent = "01:00:00";
+    final var expectedOffsetPrevious = "00:00:00";
+    final var expectedCountBefore = 1;
+
+    final var planId = merlinHelper.insertPlan(
+        missionModelId,
+        merlinHelper.admin.name(),
+        "changelog plan" + UUID.randomUUID(),
+        "1970-1-1 00:00:00+00");
+    final var activityId = merlinHelper.insertActivity(planId);
+
+    final var resBefore = connection.createStatement()
+                                    .executeQuery(
+                                        // language=sql
+                                        """
+                                            SELECT count(revision)
+                                            FROM activity_directive_changelog
+                                            WHERE plan_id = %d and activity_directive_id = %d;"""
+                                            .formatted(planId, activityId)
+                                    );
+
+    resBefore.next();
+    assertEquals(expectedCountBefore, resBefore.getInt(1));
+    resBefore.close();
+
+    final var updatedRows = connection.createStatement()
+                                      .executeQuery(
+                                          // language=sql
+                                          """
+                                              UPDATE activity_directive
+                                              SET start_offset = '%s'
+                                              WHERE plan_id = %d and id = %d
+                                              RETURNING id;"""
+                                              .formatted(expectedOffsetCurrent, planId, activityId)
```

Comment:
Rather than repeat this code, feel free to pull either `updateOffsetFromAnchor` or `updateActivityName` out into the `MerlinDBHelper` and call it here

Reply:
This snippet is now only used once

Reply:
By "repeat this code" I meant that there are already functions in the DBTests to update activities, not that this snippet appears multiple times in this file (that was a different conversation that I have already resolved).